# Video Game Analysis

## 1. Import and Clean Data

### 1a. Import Libraries and Data

In [39]:
# Import Libraries
import numpy as np
import pandas as pd

In [54]:
# Read in dataset
url = "https://raw.githubusercontent.com/kellyshreeve/Integrated_Project_1/main/moved_games.csv"
vg = pd.read_csv(url)

In [55]:
# View first 15 rows of the dataset 
vg.head(15)

Name Platform  Year_of_Release         Genre  \
0                    Wii Sports      Wii           2006.0        Sports   
1             Super Mario Bros.      NES           1985.0      Platform   
2                Mario Kart Wii      Wii           2008.0        Racing   
3             Wii Sports Resort      Wii           2009.0        Sports   
4      Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing   
5                        Tetris       GB           1989.0        Puzzle   
6         New Super Mario Bros.       DS           2006.0      Platform   
7                      Wii Play      Wii           2006.0          Misc   
8     New Super Mario Bros. Wii      Wii           2009.0      Platform   
9                     Duck Hunt      NES           1984.0       Shooter   
10                   Nintendogs       DS           2005.0    Simulation   
11                Mario Kart DS       DS           2005.0        Racing   
12  Pokemon Gold/Pokemon Silver       GB           1999.0  Role-Playing   
13                      Wii Fit      Wii           2007.0        Sports   
14           Kinect Adventures!     X360           2010.0          Misc   

    NA_sales  EU_sales  JP_sales  Other_sales  Critic_Score User_Score Rating  
0      41.36     28.96      3.77         8.45          76.0          8      E  
1      29.08      3.58      6.81         0.77           NaN        NaN    NaN  
2      15.68     12.76      3.79         3.29          82.0        8.3      E  
3      15.61     10.93      3.28         2.95          80.0          8      E  
4      11.27      8.89     10.22         1.00           NaN        NaN    NaN  
5      23.20      2.26      4.22         0.58           NaN        NaN    NaN  
6      11.28      9.14      6.50         2.88          89.0        8.5      E  
7      13.96      9.18      2.93         2.84          58.0        6.6      E  
8      14.44      6.94      4.70         2.24          87.0        8.4      E  
9      26.93      0.63      0.28         0.47           NaN        NaN    NaN  
10      9.05     10.95      1.93         2.74           NaN        NaN    NaN  
11      9.71      7.47      4.13         1.90          91.0        8.6      E  
12      9.00      6.18      7.20         0.71           NaN        NaN    NaN  
13      8.92      8.03      3.60         2.15          80.0        7.7      E  
14     15.00      4.89      0.24         1.69          61.0        6.3      E

In [56]:
# Print dataset info
vg.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16715 entries, 0 to 16714
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16713 non-null  object 
 1   Platform         16715 non-null  object 
 2   Year_of_Release  16446 non-null  float64
 3   Genre            16713 non-null  object 
 4   NA_sales         16715 non-null  float64
 5   EU_sales         16715 non-null  float64
 6   JP_sales         16715 non-null  float64
 7   Other_sales      16715 non-null  float64
 8   Critic_Score     8137 non-null   float64
 9   User_Score       10014 non-null  object 
 10  Rating           9949 non-null   object 
dtypes: float64(6), object(5)
memory usage: 1.4+ MB


This dataset has a total of 16715 entries. There are missing values in Name, Year_of_Release, Genre, Critic_Score, User_Score, and Rating. Year_of_Release needs to be changed to int data type and and User_Score needs to be changed to float data type. The variable names need to be changed to snake case.

### 1b. Rename Columns and Fix Data Types

In [57]:
# Change variable names to snake case
vg = vg.rename(
    columns={'Name':'name',
             'Platform':'platform',
             'Year_of_Release':'year_of_release',
             'Genre':'genre',
             'NA_sales':'na_sales',
             'EU_sales':'eu_sales',
             'JP_sales':'jp_sales',
             'Other_sales':'other_sales',
             'Critic_Score':'critic_score',
             'User_Score':'user_score',
             'Rating':'rating'    
})

print(vg.columns)

Index(['name', 'platform', 'year_of_release', 'genre', 'na_sales', 'eu_sales',
       'jp_sales', 'other_sales', 'critic_score', 'user_score', 'rating'],
      dtype='object')


In [72]:
# Change year_of_release to int data type

# Frist, check unique values of 'year_of_release'
print(vg['year_of_release'].unique())

<IntegerArray>
[2006, 1985, 2008, 2009, 1996, 1989, 1984, 2005, 1999, 2007, 2010, 2013, 2004,
 1990, 1988, 2002, 2001, 2011, 1998, 2015, 2012, 2014, 1992, 1997, 1993, 1994,
 1982, 2016, 2003, 1986, 2000, <NA>, 1995, 1991, 1981, 1987, 1980, 1983]
Length: 38, dtype: Int64


The values are all whole numbers, so it is safe to convert 'year_of_release' to int.

In [69]:
# Convert 'year_of_release' to int
vg['year_of_release'] = vg['year_of_release'].astype('Int64')

type = vg['year_of_release'].dtypes
print(f'The dtype for "year_of_release" now is: {type}')

The dtype for "year_of_release" now is: Int64


In [46]:
# Print unique values in 'user_score'
df['User_Score'].unique()

array(['8', nan, '8.3', '8.5', '6.6', '8.4', '8.6', '7.7', '6.3', '7.4',
       '8.2', '9', '7.9', '8.1', '8.7', '7.1', '3.4', '5.3', '4.8', '3.2',
       '8.9', '6.4', '7.8', '7.5', '2.6', '7.2', '9.2', '7', '7.3', '4.3',
       '7.6', '5.7', '5', '9.1', '6.5', 'tbd', '8.8', '6.9', '9.4', '6.8',
       '6.1', '6.7', '5.4', '4', '4.9', '4.5', '9.3', '6.2', '4.2', '6',
       '3.7', '4.1', '5.8', '5.6', '5.5', '4.4', '4.6', '5.9', '3.9',
       '3.1', '2.9', '5.2', '3.3', '4.7', '5.1', '3.5', '2.5', '1.9', '3',
       '2.7', '2.2', '2', '9.5', '2.1', '3.6', '2.8', '1.8', '3.8', '0',
       '1.6', '9.6', '2.4', '1.7', '1.1', '0.3', '1.5', '0.7', '1.2',
       '2.3', '0.5', '1.3', '0.2', '0.6', '1.4', '0.9', '1', '9.7'],
      dtype=object)

In [78]:
# Check which years are associated with user_score tbd
display(vg[vg['user_score']=='tbd'].head(30))

name platform  year_of_release  \
119                              Zumba Fitness      Wii             2010   
301             Namco Museum: 50th Anniversary      PS2             2005   
520                            Zumba Fitness 2      Wii             2011   
645                               uDraw Studio      Wii             2010   
657   Frogger's Adventures: Temple of the Frog      GBA             <NA>   
718                            Just Dance Kids      Wii             2010   
726                  Dance Dance Revolution X2      PS2             2009   
821                            The Incredibles      GBA             2004   
881              Who wants to be a millionaire       PC             1999   
1047                             Tetris Worlds      GBA             2001   
1048                          Imagine: Teacher       DS             2008   
1069                    Personal Trainer: Math       DS             2007   
1070                              Game Party 3      Wii             2009   
1089                            Monsters, Inc.      GBA             2001   
1151         SpongeBob's Atlantis SquarePantis       DS             2007   
1199                            MySims Kingdom       DS             2008   
1210              Moshi Monsters: Moshling Zoo       DS             2011   
1284                  Disney Sing It: Pop Hits      Wii             2009   
1291                      Imagine: Master Chef       DS             2007   
1292                           Disney Princess      GBA             2003   
1298           The SpongeBob SquarePants Movie      GBA             2004   
1300                   Midway Arcade Treasures      PS2             2003   
1309                                    Wall-E       DS             2008   
1341           Disney Princess: Magical Jewels       DS             2007   
1367                         Hello Kitty Party       DS             2007   
1423                             SingStar Abba      PS2             2008   
1487               Petz Wild Animals: Dolphinz       DS             2007   
1541                 Imagine: Wedding Designer       DS             2008   
1560           Mystery Case Files: MillionHeir       DS             2008   
1700                            Disney Sing It      Wii             2008   

           genre  na_sales  eu_sales  jp_sales  other_sales  critic_score  \
119       Sports      3.45      2.59      0.00         0.66           NaN   
301         Misc      2.08      1.35      0.00         0.54          61.0   
520       Sports      1.51      1.03      0.00         0.27           NaN   
645         Misc      1.65      0.57      0.00         0.20          71.0   
657    Adventure      2.15      0.18      0.00         0.07          73.0   
718         Misc      1.52      0.54      0.00         0.18           NaN   
726   Simulation      1.09      0.85      0.00         0.28           NaN   
821       Action      1.15      0.77      0.04         0.10          55.0   
881         Misc      1.94      0.00      0.00         0.00           NaN   
1047      Puzzle      1.25      0.39      0.00         0.06          65.0   
1048  Simulation      0.70      0.82      0.00         0.19           NaN   
1069      Puzzle      0.48      1.12      0.00         0.08          63.0   
1070      Puzzle      1.40      0.16      0.00         0.12          37.0   
1089   Adventure      0.59      0.95      0.03         0.10           NaN   
1151      Action      1.49      0.00      0.00         0.11          64.0   
1199  Simulation      0.80      0.59      0.01         0.16          58.0   
1210        Misc      0.37      0.96      0.00         0.21           NaN   
1284        Misc      1.07      0.28      0.00         0.13           NaN   
1291  Simulation      0.39      0.91      0.00         0.17           NaN   
1292    Platform      1.04      0.37      0.00         0.06           NaN   
1298    Platform      1.02      0.38      0.00         0.06         

Video Games with 'user_score' of tbd are all associated with jp_sales of almost zero. I will check if there are any jp_sales of almost zero that have any other user rating to confirm whether this is the reason for the rating. 

In [82]:
# Check if all jp_sales close to zero have a user_score of tbd
display(vg[(vg['jp_sales'] >= 0) & (vg['jp_sales'] <= .05)].head(30))

name platform  year_of_release  \
60                Call of Duty: Ghosts     X360             2013   
61                        Just Dance 3      Wii             2011   
66                              Halo 4     X360             2012   
68                        Just Dance 2      Wii             2010   
72                           Minecraft     X360             2013   
78                              Halo 2       XB             2004   
85                          The Sims 3       PC             2009   
89                             Pac-Man     2600             1982   
99           Call of Duty: Black Ops 3     XOne             2015   
100         Call of Duty: World at War     X360             2008   
103                         Just Dance      Wii             2009   
116                       Just Dance 4      Wii             2012   
119                      Zumba Fitness      Wii             2010   
138                  World of Warcraft       PC             2004   
139                             Driver       PS             1999   
141                      Kinect Sports     X360             2010   
143                            FIFA 15      PS4             2014   
155  LEGO Star Wars: The Complete Saga      Wii             2007   
165                 Grand Theft Auto V     XOne             2014   
170         Call of Duty: World at War      PS3             2008   
176               Assassin's Creed III     X360             2012   
179     Call of Duty: Advanced Warfare     XOne             2014   
180                          Minecraft      PS3             2014   
183                    Madden NFL 2004      PS2             <NA>   
190                     FIFA Soccer 13     X360             2012   
192                         Diablo III       PC             2012   
195         Microsoft Flight Simulator       PC             1996   
196                     Guitar Hero II      PS2             2006   
197                          Fable III     X360             2010   
205             Tony Hawk's Pro Skater       PS             1999   

            genre  na_sales  eu_sales  jp_sales  other_sales  critic_score  \
60        Shooter      6.73      2.56      0.04         0.91          73.0   
61           Misc      5.95      3.11      0.00         1.06          74.0   
66        Shooter      6.65      2.28      0.04         0.74          87.0   
68           Misc      5.80      2.85      0.01         0.78          74.0   
72           Misc      5.70      2.65      0.02         0.81           NaN   
78        Shooter      6.82      1.53      0.05         0.08          95.0   
85     Simulation      0.99      6.42      0.00         0.60          86.0   
89         Puzzle      7.28      0.45      0.00         0.08           NaN   
99        Shooter      4.59      2.11      0.01         0.68           NaN   
100       Shooter      4.81      1.88      0.00         0.69          84.0   
103          Misc      3.48      2.99      0.00         0.73          49.0   
116          Misc      4.05      2.16      0.00         0.55          74.0   
119        Sports      3.45      2.59      0.00         0.66           NaN   
138  Role-Playing      0.08      6.21      0.00         0.00          93.0   
139        Action      3.11      2.80      0.02         0.33          87.0   
141        Sports      3.92      1.73      0.03         0.51          73.0   
143        Sports      0.80      4.33      0.05         0.90          82.0   
155        Action      3.57      1.55      0.00         0.51          80.0   
165        Action      2.81      2.19      0.00         0.47          97.0   
170       Shooter      2.73      1.83      0.00         0.83          85.0   
176        Action      3.13      1.69      0.03         0.44          84.0   
179       Shooter      3.22      1.55      0.01         0.48          81.0   
180          Misc      2.03      2.37      0.00         0.87           NaN   
183        Sports      4.26      0.26      0.01      

There are user_scores present for other instances of jp_sales close to zero. It does not appear that these sales figures are the reason for the tbd rating. There do not appear to be any other patterns in year, genre, sales, critic_score, or rating that would explain the user_score of tbd. Because there are no clear patterns explaining this value, I will treat tbd as a missing value. 

In [ ]:
# Fill user_score tbd with nan


### 1c. Address Missing Values

### 1d. Check for Duplicates

### 1e. Add Additional Features

## 2. Analyze the data

## 3. Create a User Profile for Each Region

## 4. Test Hypotheses

## 5. Conclusion